electroncash_plugins/fusion/qt.py

```diff
@@ -618,9 +619,14 @@ def on_status(controller):
     @hook
     def history_list_filter(self, history_list, h_item, label):
         # NB: 'h_item' might be None due to performance reasons
-        if self._hide_history_txs:
-            return bool(label.startswith("CashFusion "))  # this string is not translated for performance reasons
-        return None
+        if h_item is None or not self._hide_history_txs:
+            return None
+
+        txid = h_item.tx_hash
+        if not txid in self._is_fuz_tx:
+            self._is_fuz_tx[txid] = FusionPlugin.check_is_fuz_tx(history_list.wallet, txid)
+
+        return self._is_fuz_tx[txid]
```

Comment:
Well that `check_is_fuz_tx` function can return `None` which signifies that it couldn't (yet) find the tx for some reason in the wallet.  It's a theoretical corner case due to the API contract of the function, so it should be handled since it is a branch that exists.  As such, we shouldn't blindly store whatever it returns, but only store if not `None`.

The below also avoids double-lookup in the dict in the happy case where there is a cache hit.

```suggestion
        retval = self._is_fux_tx.get(txid)
        if retval is None:
            retval = FusionPlugin.check_is_fuz_tx(history_list.wallet, txid)
            if retval is not None:
                self._is_fuz_tx[txid] = retval

        return retval
```